### PR TITLE
Actually working tests for Spree::Store.default

### PR DESCRIPTION
**Pull request by jrochkind**
The previous tests weren't actually testing what they were described to test,
they weren't testing anything but their own setup.

I think these tests effectively test what was meant to be tested.

### CODE REVIEW

**Comment by jordan-brough** [comment]
Thanks!
👍 (assuming specs go green)

File: core/spec/models/spree/store_spec.rb [hunk 1]
```diff
@@ -37,19 +37,32 @@
   end
 
   describe ".default" do
-    let!(:store)    { create(:store) }
-    let!(:store_2)  { create(:store, default: true) }
+    before do
+      Spree::Store.delete_all
```
Comment (jhawthorn) [inline core/spec/models/spree/store_spec.rb]: This is unnecessary. There aren't any stores being created before this (and we truncate the database at the start of the core test suite)

Otherwise :+1:

Comment (jrochkind) [inline core/spec/models/spree/store_spec.rb]: yep, I thought it might be, but it's so easy to mess this test up so it's not testing what you want (as it was before), that I thought it good to put in the explicit delete to make absolutely sure it's testing what it's supposed to. 

Especially because it requires macro knowledge of what's going on in the test suite and solidus code (some code that at some point has lazily created a default store?) to be _sure_ there aren't any stores created before.  Truncating at the start of the test suite isn't enough, you also need to be certain you have something running before each test that makes sure the db is being reset. Which there should be -- but I still don't understand how all the _other_ tests that require a default store existing actually work in that case!  

There is so much macro knowledge you need to be sure, and so many things that could change to make it not so, and such risk of the test not testing what is intended,  it just seemed safer to explicitly enforce the expectations instead of just counting on everything else working how you expect. 

But if you still don't after this persuasive essay, I'll get rid of it. Let me know?

Comment (jhawthorn) [inline core/spec/models/spree/store_spec.rb]: We need to be able to assume that we have a reasonable test framework and that the database is empty at the start of our tests. Otherwise we'd need to put `delete_all`s before practically every test we have.

Comment (jrochkind) [inline core/spec/models/spree/store_spec.rb]: okeydoke, I'll delete it. I still have no idea how a bunch of other tests (especially controller/feature tests) are managing to pass _without_ any apparent creation of a spree default store, now that orders require stores. I guess something must be creating a spree default store somewhere else I haven't noticed.

**Comment by jrochkind** [comment]
@jhawthorn removed the unneccesary delete_all. thanks for feedback!

**Comment by mamhoff** [comment]
👍

File: core/spec/models/spree/store_spec.rb (hunk 1 above)
Comment (jordan-brough) [inline core/spec/models/spree/store_spec.rb]: > I still have no idea how a bunch of other tests (especially controller/feature tests) are managing to pass without any apparent creation of a spree default store

@jrochkind are there any in particular that you see that look weird? I was curious about this also so I took a look at a few specs and the ones I looked at all seemed to be creating a default store in one way or another:

- [frontend/spec/features/checkout_spec.rb](https://github.com/solidusio/solidus/blob/3a8592d6861f12a9edd063858652729facfde377/frontend/spec/features/checkout_spec.rb#L4) (via ["checkout setup"](https://github.com/solidusio/solidus/blob/3a8592d6861f12a9edd063858652729facfde377/frontend/spec/support/shared_contexts/checkout_setup.rb))
- [frontend/spec/features/order_spec.rb](https://github.com/solidusio/solidus/blob/3a8592d6861f12a9edd063858652729facfde377/frontend/spec/features/order_spec.rb#L4) (via [OrderWalkthrough](https://github.com/solidusio/solidus/blob/3a8592d6861f12a9edd063858652729facfde377/core/lib/spree/testing_support/order_walkthrough.rb#L23))
- [api/spec/controllers/spree/api/orders_controller_spec.rb](https://github.com/solidusio/solidus/blob/3a8592d6861f12a9edd063858652729facfde377/api/spec/controllers/spree/api/orders_controller_spec.rb#L7) (via [order factory](https://github.com/solidusio/solidus/blob/3a8592d6861f12a9edd063858652729facfde377/core/lib/spree/testing_support/factories/order_factory.rb#L15))
- [backend/spec/controllers/spree/admin/orders_controller_spec.rb](https://github.com/solidusio/solidus/blob/3a8592d6861f12a9edd063858652729facfde377/backend/spec/controllers/spree/admin/orders_controller_spec.rb#L6)

**Comment by jhawthorn** [comment]
:+1: Thanks again
